pkg/pillar/hypervisor/kubevirt.go

```diff
@@ -1314,8 +1314,10 @@ func InfoReplicaSetContainer(ctx kubevirtContext, repName string) (string, error
 	if !ok {
 		return "", logError("Failed to get nodeName")
 	}
+
 	pods, err := podclientset.CoreV1().Pods(kubeapi.EVEKubeNameSpace).List(context.TODO(), metav1.ListOptions{
 		LabelSelector: fmt.Sprintf("app=%s", repName),
+		FieldSelector: fmt.Sprintf("spec.nodeName=%s", nodeName),
```

Comment:
there is a 'NonLocal' logic w/ this pods, w/ this change, maybe the non-local needs to be removed?

Reply:
Not clear on non-local needs to be removed

Reply:
basically what the previous code was that, it does gets 'any' pod which belongs to the replicaset, may or may not be on this particular node. but the next for loop is to find if not running on this node, and label it as 'non-local', and continue. So if you have to filter only pick on this node, then this 'non-local' code is not needed.

Reply:
ok I will remove that.

Reply:
You mean this code

		if nodeName != pod.Spec.NodeName {
			foundNonlocal = true
			logrus.Infof("InfoReplicaSetContainer: rep %s, nodeName %v differ w/ hostname", repName, pod.Spec.NodeName)
			continue
		}


and

if foundNonlocal {
		return "NonLocal", nil
	}

Reply:
yes.

Reply:
Amended the commit, please check